app/src/main/java/ru/practicum/android/diploma/search/ui/SearchFragment.kt

```diff
@@ -0,0 +1,80 @@
+package ru.practicum.android.diploma.search.ui
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import ru.practicum.android.diploma.databinding.FragmentSearchBinding
+import ru.practicum.android.diploma.search.presentation.SearchViewModel
+import ru.practicum.android.diploma.search.presentation.models.UiScreenState
+import ru.practicum.android.diploma.search.presentation.models.VacancyUi
+
+class SearchFragment : Fragment() {
+    private var binding: FragmentSearchBinding? = null
+    private val viewModel by viewModel<SearchViewModel>()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        binding = FragmentSearchBinding.inflate(inflater, container, false)
+        return binding?.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        viewModel.uiState.observe(viewLifecycleOwner) {
+            renderUiState(it)
+        }
+    }
+
+    private fun renderUiState(state: UiScreenState) {
+        when (state) {
+            UiScreenState.Default -> showDefaultState()
+            UiScreenState.Empty -> showEmptyState()
+            UiScreenState.Loading -> showLoadingState()
+            UiScreenState.NoInternetError -> showNoInternetErrorState()
+            UiScreenState.ServerError -> showServerErrorState()
+            is UiScreenState.Success -> showSuccessState(state.vacancies, state.found)
+        }
+    }
+
+    @Suppress("EmptyBlock")
+    private fun showDefaultState() {
+        // Implement this method later
+    }
```

Comment:
Вместо того, чтобы suppress'ом затыкать детект
```suggestion
    private fun showDefaultState() = Unit
```